### ISSUE TITLE
File attachment on emails

### ISSUE THREAD

**Issue by brmesquita** [opensupports/opensupports#494]
Hi, I was testing out the ticket creation from IMAP and it works great, except for file attachments. Ticket is created, all is well, but without attachments.
Running through the code, some of it is commented out and it even deletes the temp files.
Isn't file attachement supported? Will it be? What's the workaround?

**Comment by nickbe** [comment]
I don't think it already is for now. Ticket by mail - especially with attachments are tricky anyway. Even more with attachments (although this feature would be cool).

**Comment by brmesquita** [comment]
Creating tickets by email is working fine! Just tested it with the latest version and worked great!
The only downside is that file attachments are discarded, although the code is there (at least partially).
This one functionality is the "make it or break it" for me, really need it.
Loving all the rest! Devs, are there any updates?

**Comment by brmesquita** [comment]
@(Awesome Devs), is there something I can do to make it work? Is it implemented or is there any patch?

**Comment by TobiZwo** [comment]
> 
> 
> Creating tickets by email is working fine! Just tested it with the latest version and worked great!
> The only downside is that file attachments are discarded, although the code is there (at least partially).
> This one functionality is the "make it or break it" for me, really need it.
> Loving all the rest! Devs, are there any updates?

Unfortunately creating tickets by email is not working fine. Tickets are created with correct title and author but with no content. Is it only me who is affected? I am using V4.4. with GMX Email.

**Comment by brmesquita** [comment]
@TobiZwo A lot has been changed on my side since then.
Ticket creation from mails is working on my side but I patched a lot of situations, like:
- Text emails were 100%, but html emails behaved exactly as you said
- File attachments were discarded fully (although part of the code was on the platform)
- There are issues between mail sender and platform user that made some tickets be discarded alltogether

I patched a lot of this situations on my side and have it working now with all emails, users and file attachments. As OS only allows one file per ticket, I made it zip all files attachments and insert that zip on the ticket.
But a lot of information is still hardcoded (email headers, some language strings) and the developers still have some work to do for a fully production version.

You can also do it fiddling with the code, its not that hard, but cant do it "out of the box"

**Comment by mredigonda** [comment]
Thanks for reporting this, yes, we still need to work harder for a better email integration, which is crucial for a ticketing system.

Feel free to send a pull request with your patches, even if they are not complete or have any kind of hardcoding, it's going to be useful for us to get to a fully working solution.
